### PR TITLE
feat(cra-metrics): Process metrics crash rate alerts updates [INGEST-630]

### PR DESCRIPTION
**Pull request by ahmedetefy**
Adds functionality that processes updates from metrics crash rate alerts subscription updates

### CODE REVIEW

File: src/sentry/incidents/subscription_processor.py [hunk 1]
```diff
@@ -259,10 +260,61 @@ def get_crash_rate_alert_aggregation_value(subscription_update):
...
+
     def get_aggregation_value(self, subscription_update):
-        is_sessions_dataset = Dataset(self.subscription.snuba_query.dataset) == Dataset.Sessions
-        if is_sessions_dataset:
+        if self.subscription.snuba_query.dataset == QueryDatasets.SESSIONS.value:
             aggregation_value = self.get_crash_rate_alert_aggregation_value(subscription_update)
+        elif self.subscription.snuba_query.dataset == QueryDatasets.METRICS.value:
+            aggregation_value = self.get_crash_rate_alert_metrics_aggregation_value(
+                subscription_update
+            )
```
Comment (jjbayer) [inline src/sentry/incidents/subscription_processor.py:317]: Does this mean that any subscription update received for the metrics dataset will be treated as a session crash rate result? If yes, I guess that's fine for now, but we will potentially have many different use cases for metrics-based alerts in the future.

Comment (untitaker) [reply src/sentry/incidents/subscription_processor.py:317]: don't we get some query ID back from snuba on which we can dispatch?

**Review by untitaker — APPROVED** [review]

File: src/sentry/incidents/subscription_processor.py (hunk 1 above)
Comment (ahmedetefy) [reply src/sentry/incidents/subscription_processor.py:317]: Yep we get a `subscription_id` that can be used for that purpose I guess
fwiw this is what the payload returned looks like 
```
{'subscription_id': '0/b2ec54a04de411ecb2e1acde48001122', 'request': {'query': 'MATCH (metrics_counters) SELECT sum(value) AS value BY project_id, tags[3] WHERE org_id = 1 AND project_id IN array(1) AND metric_id = 7', 'parent_api': '<unknown>'}, 'result': {'data': [{'project_id': 1, 'tags[3]': 13, 'value': 8.0}, {'project_id': 1, 'tags[3]': 4, 'value': 46.0}], 'meta': [{'name': 'project_id', 'type': 'UInt64'}, {'name': 'tags[3]', 'type': 'UInt64'}, {'name': 'value', 'type': 'Float64'}]}, 'timestamp': datetime.datetime(2021, 11, 25, 13, 1, tzinfo=<UTC>), 'values': {'data': [{'project_id': 1, 'tags[3]': 13, 'value': 8.0}, {'project_id': 1, 'tags[3]': 4, 'value': 46.0}], 'meta': [{'name': 'project_id', 'type': 'UInt64'}, {'name': 'tags[3]', 'type': 'UInt64'}, {'name': 'value', 'type': 'Float64'}]}}
```

**Review by jjbayer — APPROVED** [review]

File: src/sentry/incidents/subscription_processor.py (hunk 1 above)
Comment (wedamija) [reply src/sentry/incidents/subscription_processor.py:317]: If you want these to do different things then that could live on the `AlertRule` itself, which has a link to the subscription via the `subscription_id`

**Pull request merged**